### DISCUSSION TITLE
Is there a way to have a long time access token

### DISCUSSION THREAD

**Discussion by p2cardon** [TeslaBuds/AuthAppForTesla#18] [Q&A]
I'm using your app on PowerBook Pro M1, and it's working pretty well …
The provided Access token is valid for about 8 hours … is there a way to get a long time (45 days)
 access token as some other applications do ?

Best regards
Pascal

**Comment by Kemmey** [comment]
Hey,

Yeah, you Can get the old v2 access token using the shortcut action. I removed it from the main app ui, as Tesla has deprecated the v2 tokens, and I didn't want to confuse people as to which token they should use. 

You definitely should use the v3 token. 

Alternatively, grab the refresh token and write your own logic to get a v3 access token whenever needed. 

Note the v3 access token auto renews when using the shortcuts action.

**Reply by rileymd88** [reply]
Hi @Kemmey - Can you link me to where you are actually generating the V3 tokens I am a bit confused on how to do this and I do not think it is documented anywhere - for example if I follow the documentation here: https://tesla-api.timdorr.com/api-basics/authentication I end up getting a V2 token.

Thanks!

**Reply by Kemmey** [reply]
https://github.com/TeslaBuds/AuthAppForTesla/blob/main/AuthAppForTesla/AuthController.swift

**Reply by Kemmey** [reply]
Look for acquire silent
